packages/web3-core-requestmanager/src/index.ts

```diff
@@ -0,0 +1,51 @@
+import Web3ProviderHttp from 'web3-providers-http';
+import { ProviderOptions } from 'web3-providers-base/types';
+
+// TODO Make eslint happy
+/* eslint-disable */
+enum ProviderProtocol { UNKNOWN, HTTP, WS, IPC }
+
+export default class Web3RequestManager {
+    provider: Web3ProviderHttp | undefined
+
+    providerProtocol: ProviderProtocol = ProviderProtocol.UNKNOWN
+
+    constructor(providerOptions: ProviderOptions) {
+      switch (Web3RequestManager.detectProviderProtocol(providerOptions.providerString)) {
+        case ProviderProtocol.HTTP:
+          this.provider = new Web3ProviderHttp(providerOptions);
+          this.providerProtocol = ProviderProtocol.HTTP;
+          break;
+        case ProviderProtocol.WS:
+          this.providerProtocol = ProviderProtocol.WS;
+          // TODO
```

Comment:
might be good practice to `throw` here so that we remember it doesn't exist

Reply:
Code now throws for `WS`, `IPC`, and `UNKNOWN`